go.mod

```diff
@@ -1,15 +1,17 @@
 module github.com/google/osv-scanner
 
-go 1.21
+go 1.21.0
```

Comment:
```suggestion
go 1.21
```

Reply:
Just removing the .0 should make osv-scanner assume it's the latest version with #740

Reply:
This change was required by Go - it demanded I run `go mod tidy` and this was the result 🤷‍♂️

I think it might be related to `go-snaps`?

Reply:
ah, I think it might just be go1.21 compiler as well. Let's update this back to `1.21.5` then, since we only support go 1.21 to compile osv-scanner anyway, not supporting older compilers doesn't matter too much here.